lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -3693,22 +3693,63 @@ NormalProtocolConformance *GetImplicitSendableRequest::evaluate(
   }
 
   // Local function to form the implicit conformance.
-  auto formConformance = [&]() -> NormalProtocolConformance * {
+  auto formConformance = [&](const DeclAttribute *attrMakingUnavailable)
+        -> NormalProtocolConformance * {
     ASTContext &ctx = nominal->getASTContext();
     auto proto = ctx.getProtocol(KnownProtocolKind::Sendable);
     if (!proto)
       return nullptr;
 
+    DeclContext *conformanceDC = nominal;
+    if (attrMakingUnavailable) {
+      llvm::VersionTuple NoVersion;
+      auto attr = new (ctx) AvailableAttr(SourceLoc(), SourceRange(),
+                                          PlatformKind::none, "", "", nullptr,
+                                          NoVersion, SourceRange(),
+                                          NoVersion, SourceRange(),
+                                          NoVersion, SourceRange(),
+                                          PlatformAgnosticAvailabilityKind::Unavailable,
+                                          false);
+
+      // Conformance availability is currently tied to the declaring extension.
+      // FIXME: This is a hack--we should give conformances real availability.
+      auto inherits = ctx.AllocateCopy(makeArrayRef(
+          InheritedEntry(TypeLoc::withoutLoc(proto->getDeclaredInterfaceType()),
+                         /*isUnchecked*/true)));
+      auto extension = ExtensionDecl::create(ctx, attrMakingUnavailable->AtLoc,
```

Comment:
You might be able to hook this into a lower level in `ConformanceLookupTable` and create the conformance there, instead of actually synthesizing an ExtensionDecl.

Reply:
For now, I think this way is actually better because we can print the extension into module interfaces through normal paths. If we eventually introduce a new syntax for applying availability directly to a conformance, we should probably drop this and switch over to using that instead.